GdscBackend/Controllers/v1/TechnologyController.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Net.Mime;
+using System.Threading.Tasks;
+using gdsc_web_backend.Database;
+using gdsc_web_backend.Models;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace gdsc_web_backend.Controllers.v1
+{
+    [ApiController]
+    [ApiVersion("1")]
+    [Route("api/v1/technologies")]
+    [Consumes(MediaTypeNames.Application.Json)] 
+    [Produces(MediaTypeNames.Application.Json)] 
+    public class TechnologyController : ControllerBase
+    {
+        private readonly IRepository<TechnologyModel> _repository;
+        public TechnologyController(IRepository<TechnologyModel> repository)
+        {
+            _repository = repository;
+        }
+        
+        //get all
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<IEnumerable<TechnologyModel>>> GetAll()
+        {
+            return Ok((await _repository.GetAsync()).ToList());
+        }
+        
+        //post
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<ActionResult<TechnologyModel>> Post(TechnologyModel entity)
+        {
+            entity = await _repository.AddAsync(entity);
+            return CreatedAtAction(nameof(Post), new {entity.Id}, entity);
+        }
+        
+        //delete
```

Comment:
There should not be comment that are obvious. We understand what is happening from the method name